lxc/project.go

```diff
@@ -0,0 +1,539 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"syscall"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/api"
+	cli "github.com/lxc/lxd/shared/cmd"
+	"github.com/lxc/lxd/shared/i18n"
+	"github.com/lxc/lxd/shared/termios"
+)
+
+type cmdProject struct {
+	global *cmdGlobal
+}
+
+func (c *cmdProject) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = i18n.G("project")
+	cmd.Short = i18n.G("Manage projects")
+	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
+		`Manage projects`))
+
+	// Create
+	projectCreateCmd := cmdProjectCreate{global: c.global, project: c}
+	cmd.AddCommand(projectCreateCmd.Command())
+
+	// Delete
+	projectDeleteCmd := cmdProjectDelete{global: c.global, project: c}
+	cmd.AddCommand(projectDeleteCmd.Command())
+
+	// Edit
+	projectEditCmd := cmdProjectEdit{global: c.global, project: c}
+	cmd.AddCommand(projectEditCmd.Command())
+
+	// List
+	projectListCmd := cmdProjectList{global: c.global, project: c}
+	cmd.AddCommand(projectListCmd.Command())
+
+	// Rename
+	projectRenameCmd := cmdProjectRename{global: c.global, project: c}
+	cmd.AddCommand(projectRenameCmd.Command())
+
+	// Show
+	projectShowCmd := cmdProjectShow{global: c.global, project: c}
+	cmd.AddCommand(projectShowCmd.Command())
+
+	// Set default
+	projectSwitchCmd := cmdProjectSwitch{global: c.global, project: c}
+	cmd.AddCommand(projectSwitchCmd.Command())
+
+	return cmd
+}
+
+// Create
+type cmdProjectCreate struct {
+	global  *cmdGlobal
+	project *cmdProject
+}
+
+func (c *cmdProjectCreate) Command() *cobra.Command {
+	cmd := &cobra.Command{}
+	cmd.Use = i18n.G("create [<remote>:]<project>")
+	cmd.Short = i18n.G("Create projects")
+	cmd.Long = cli.FormatSection(i18n.G("Description"), i18n.G(
+		`Create projects`))
+
+	cmd.RunE = c.Run
+
+	return cmd
+}
+
+func (c *cmdProjectCreate) Run(cmd *cobra.Command, args []string) error {
+	// Sanity checks
+	exit, err := c.global.CheckArgs(cmd, args, 1, 1)
+	if exit {
+		return err
+	}
+
+	// Parse remote
+	resources, err := c.global.ParseServers(args[0])
+	if err != nil {
+		return err
+	}
+
+	resource := resources[0]
+
+	if resource.name == "" {
+		return fmt.Errorf(i18n.G("Missing project name"))
+	}
+
+	// Create the project
+	project := api.ProjectsPost{}
+	project.Name = resource.name
+	project.Config = map[string]string{}
+	project.Config["features.images"] = "true"
+	project.Config["features.profiles"] = "true"
```

Comment:
Changing those can be done through `lxc project edit` but the code above is still not what we should do.

Instead we should just keep project.Config set to `nil` and have the server set all feature to `true` in that case, that's so we don't need the client to know what features may or may not be available on a given server.

Reply:
Actually, since we've turned the features into config options, we also do need to add the usual `get`, `set` and `unset` sub-commands to `lxc project` and we should also allow the `-c` option to `lxc project create` for good measure so that the user can set those at creation time.

@freeekanayaka

Reply:
Should be all done (new commands and default feature values).